pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -336,9 +336,13 @@ func dryRunPreemption(ctx context.Context, fh framework.Handle,
 		stateCopy := state.Clone()
 		pods, numPDBViolations, status := selectVictimsOnNode(ctx, fh, stateCopy, pod, nodeInfoCopy, pdbs)
 		if status.IsSuccess() {
-			victims := extenderv1.Victims{
-				Pods:             pods,
-				NumPDBViolations: int64(numPDBViolations),
+			var victims extenderv1.Victims
+			// do not create victims without pods
+			if len(pods) != 0 {
+				victims = extenderv1.Victims{
+					Pods:             pods,
+					NumPDBViolations: int64(numPDBViolations),
+				}
```

Comment:
We should treat `len(pods) == 0` as an internal error, right? Why not just do `if status.IsSuccess() && len(pods) != 0` in L338?

Reply:
That's my original version but the unit test failed as there are test cases that return empty victims with fit= true. Please see my comment. https://github.com/kubernetes/kubernetes/pull/101560#issuecomment-830683854

Reply:
yes, please update those tests.

Reply:
should  ` status.IsSuccess() && len(pods) == 0`  be considered an error or just ignored?

Reply:
let's craft it as an internal error.